### PR TITLE
Project: document release process

### PR DESCRIPTION
**Pull request by vbatts**
Fairly simple. Docs for various project processes, starting with release process.

### CODE REVIEW

File: project.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Project docs
+
+## Release Process
+
+* Increment version in version.go
```
Comment (wking) [inline project.md]: Does this happen before or after the release?  See #203 bumping well before the 0.2.0 release, and my [comment][1] in #194 suggesting we either use SemVer's pre-release markers or increase our cadence.

[1]: https://github.com/opencontainers/specs/pull/194#issuecomment-142062583

**Comment by wking** [comment]
Release notes might be a useful addition to this list.  It's more
efficient to have a few people summarize the changes
(backwards-incompatible, extensions, clarifications, …) between v0.1.0
and v0.2.0 than to have all spec-readers dig through the Git history.

**Comment by tianon** [comment]
LGTM

File: project.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+* `git commit` version increment
+* `git tag` the prior commit (preferrably signed tag)
+* `make docs` to produce PDF and HTML copies of the spec
+* Make a release on [github.com/opencontainers/specs](https://github.com/opencontainers/specs/releases) for the version. Attach the produced docs.
```
Comment (mrunalp) [inline project.md]: Move the last sentence to a new line.

Comment (vbatts) [inline project.md]: it's on a bullet though

Comment (philips) [inline project.md]: Yea, disregard the newline thing. Our team actually deprecated the one sentence per-line rule after finding that `git log -p --word-diff` and github's improved "smart diff" is sufficient: https://github.com/coreos/docs/blob/master/STYLE.md#line-wrapping-considered-harmful

**Comment by mrunalp** [comment]
@philips Thanks :) LGTM

**Pull request merged**